strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -203,20 +203,18 @@ public void deleteFromIndex(RepositoryPath path)
             throws IOException
     {
         Repository repository = path.getFileSystem().getRepository();
-        RepositoryFileAttributes a = (RepositoryFileAttributes) Files.readAttributes(path, BasicFileAttributes.class);
 
-        if (!repositoryFeatures.isIndexingEnabled(repository) || a.isMetadata())
+        if (!repositoryFeatures.isIndexingEnabled(repository) || isMetadata(path.toString()))
```

Comment:
why is this changed? we have common way to do it thru `RepositoryFileAttributes`.

Reply:
Because 

`RepositoryFileAttributes a = (RepositoryFileAttributes) Files.readAttributes(path, BasicFileAttributes.class);`

requires physicall existence of the file, but in terms of group repositories there won't be any artifacts.

`deleteFromIndex` should also work for group repositories

Reply:
right,
thanks for the explanation
maybe `RepositoryFileAttributes` should work for non-existent files too, I will look at this in future